### PR TITLE
Added canwear var to pets, in order to prevent invisible pets

### LINKED ISSUE #33142: Putting pet collars on gondolas makes them invisible
Step 1: Make a gondola
Step 2: Put a pet collar on it
Step 3: Have invisible gondola sneak up on people and nuzzle them without them knowing what's going on

### PR DESCRIPTION
**Pull request by Wad67**
What it says on the tin.
All current pets bar gondola can wear collars, & tags.

Put in place to prevent pets from going invisible if they don't have the correct icon state.
Fixes #33142

### CODE REVIEW

File: code/modules/mob/living/simple_animal/friendly/gondola.dm [hunk 1]
```diff
@@ -25,6 +25,8 @@
 	maxHealth = 200
 	health = 200
 	del_on_death = TRUE
+	//cannot wear pet collars and tags because nobody has made relevant icons
+	canwear = 0
```
Comment (Cyberboss) [inline code/modules/mob/living/simple_animal/friendly/gondola.dm]: FALSE

File: code/modules/mob/living/simple_animal/friendly/pet.dm [hunk 2]
```diff
@@ -4,21 +4,23 @@
 	var/obj/item/clothing/neck/petcollar/pcollar = null
 	var/collar = ""
 	var/pettag = ""
+	var/canwear = 1
```
Comment (Cyberboss) [inline code/modules/mob/living/simple_animal/friendly/pet.dm]: TRUE

**Review by Cyberboss — CHANGES_REQUESTED** [review]

File: code/modules/mob/living/simple_animal/friendly/pet.dm (hunk 2 above)
Comment (lzimann) [inline code/modules/mob/living/simple_animal/friendly/pet.dm]: this var should be named `can_wear_pet_collar` or something similar, canwear is too ambiguous and vague

**Comment by Wad67** [comment]
Will perform the suggested changes, 
Assuming we are using TRUE & FALSE respectively for code readability?

**Comment by Cyberboss** [comment]
Yes

**Review by Cyberboss — DISMISSED** [review]

File: code/modules/mob/living/simple_animal/friendly/pet.dm [hunk 3]
```diff
@@ -4,21 +4,23 @@
...
-		collar = "[icon_state]collar"
-		pettag = "[icon_state]tag"
-		regenerate_icons()
-		to_chat(user, "<span class='notice'>You put the [P] around [src]'s neck.</span>")
-		if(P.tagname)
-			real_name = "\proper [P.tagname]"
-			name = real_name
-		qdel(P)
-		return
+		if(can_wear_pet_items)
```
Comment (ShizCalev) [inline code/modules/mob/living/simple_animal/friendly/pet.dm]: Unnest this.

Comment (ShizCalev) [inline code/modules/mob/living/simple_animal/friendly/pet.dm]: Unnest this.

**Review by ShizCalev — CHANGES_REQUESTED** [review]

**Comment by AnturK** [comment]
If this is only meant to prevent the missing icons instead of being editable just check if the states exist ?

File: code/modules/mob/living/simple_animal/friendly/pet.dm (hunk 3 above)
Comment (Wad67) [reply code/modules/mob/living/simple_animal/friendly/pet.dm]: Not immediately obvious what you want un-nested , are you referring to if(P.tagname) ?
Should only apply a name if the pet can receive a collar.

**Comment by Wad67** [comment]
That could most certainly be done in lieu of applying a var. 
If I can get that working, I'll update this pr.

**Comment by Wad67** [comment]
Shit, I think the gitignore did not apply for some reason

**Comment by Wad67** [comment]
Going to close this because I fucked up.

**Pull request closed without merging**